react-native/react/native/remote-manager.desktop.js

```diff
@@ -34,29 +34,56 @@ class RemoteManager extends Component {
     }
   }
 
-  componentWillReceiveProps (nextProps) {
-    if (!this.props.trackerServerActive && nextProps.trackerServerActive) {
-      this.setState({showTrackerPopup: true})
-    }
+  windowStates (trackers) {
+    return Object.keys(trackers).map(user => {
+      return `${user}:${trackers[user].closed ? 0 : 1}`
+    }).join(',')
   }
 
-  render () {
-    if (!this.props.trackerClosed) {
-      return (
-        <RemoteComponent
-          windowsOpts={{
-            height: 332,
-            width: 520,
-            frame: false,
-            resizable: false
-          }}
-          waitForState
-          onRemoteClose={this.props.onCloseFromHeader}
-          component='tracker'/>
-      )
+  shouldComponentUpdate (nextProps, nextState) {
+    // different window states
+    if (this.windowStates(nextProps.trackers) !== this.windowStates(this.props.trackers)) {
+      return true
     }
 
-    return (<div/>)
+    return false
+  }
+
+  componentWillReceiveProps (nextProps) {
+    let popups = {}
+
+    Object.keys(nextProps.trackers).forEach(username => {
+      if (!this.state.popups[username]) {
+        popups[username] = (
+          <RemoteComponent
+            windowsOpts={{
+              height: 332,
+              width: 520,
+              frame: false,
+              resizable: false
+            }}
+            waitForState
+            onRemoteClose={this.props.onCloseFromHeader}
```

Comment:
shouldn't you have a key attribute here so react can determine which one is for which?